src/VisualStudio/Xaml/Impl/Implementation/XamlTextViewCreationListener.cs

```diff
@@ -100,25 +98,19 @@ public void VsTextViewCreated(IVsTextView vsTextView)
                 return;
             }
 
-            AbstractProject project = GetXamlProject(hierarchy);
-            if (project == null)
+            VisualStudioProject project;
+
+            if (!_xamlProjects.TryGetValue(hierarchy, out project))
             {
-                project = new XamlProject(
-                    _vsWorkspace.GetProjectTrackerAndInitializeIfNecessary(_serviceProvider),
-                    hierarchy,
-                    _serviceProvider,
-                    _vsWorkspace);
+                string name;
+                hierarchy.TryGetName(out name);
+                project = _visualStudioProjectFactory.CreateAndAddToWorkspace($"{name}-XamlProject", StringConstants.XamlLanguageName);
```

Comment:
>$"{ [](start = 78, length = 3)

Nit: `name + "-XamlProject"` would do.

Reply:
Any concern with uniqueness of the name? What about a workspace with projects `Foo` and `Foo-XamlProject`?

---
In reply to: [216371648](https://github.com/dotnet/roslyn/pull/26931#discussion_r216371648) [](ancestors = 216371648)

Reply:
This line moved from XamlProject.cs, so the choice of project name isn't new. I could imagine the designer depends on it, so won't fix this here. Will fix your nit though.